src/Chart.ts

```diff
@@ -0,0 +1,130 @@
+import {DataFiltererOptions, FilteredData, MeanData} from "./DataFilterer";
+import {Chart, ChartConfiguration} from "chart.js";
+
+export interface CustomChartOptions extends DataFiltererOptions {
+    plotTitle: string;
+    yAxisTitle: string;
+    currentPlot: string;
+    hideLabels: boolean;
+}
+
+function rescaleLabel(value: number, scale: number): string {
+    if (scale > 1000000000) {
+        return value / 1000000000 + "B";
+    }
+    if (scale > 1000000) {
+        return value / 1000000 + "M";
+    }
+    if (scale > 1000) {
+        return value / 1000 + "K";
+    }
+    return value.toString();
+}
+
+export function impactChartConfig(filterData: FilteredData,
```

Comment:
I've split the chart configuration logic out into it's own file, because I was finding `data_vis.ts` a bit unwieldy. I also found that factoring out this logic helped me see exactly what inputs were needed, which made it easier to reason about when the graph would update.